### ISSUE TITLE
Using .join() on an empty Vec stalls

### ISSUE THREAD

**Issue by michael-temp** [yoshuawuyts/futures-concurrency#153]
Is this the intended behaviour?

```toml
futures-concurrency = "7.4.1"
```

```rust
    pub use futures_concurrency::prelude::*;

    #[tokio::test]
    // #[ignore]
    async fn test_join_empty_vec() -> Result<()> {
        Vec::<AbortOnDrop<()>>::new().join().await;

        Ok(())
    }
```

Result:

```
running 1 test
        SLOW [>  5.000s] tokio_fixes::tests::test_join_empty_vec
```

**Comment by yoshuawuyts** [comment]
You're right that this is not the intended behavior. I've posted a patch for it in https://github.com/yoshuawuyts/futures-concurrency/pull/154. Thank you for reporting this!

**Comment by yoshuawuyts** [comment]
Fixed in `v7.4.2`
